hub/core/tensor.py

```diff
@@ -1,29 +1,33 @@
-from hub.core.index import Index
+import re
 from typing import List, Tuple, Union
-import numpy as np
 
+import exiftool  # type: ignore
+import numpy as np
 from hub.core.chunk_engine.read import sample_from_index_entry
 from hub.core.chunk_engine.write import write_bytes
+from hub.core.flatten import row_wise_to_bytes
+from hub.core.index import Index
 from hub.core.meta.index_map import read_index_map, write_index_map
 from hub.core.meta.tensor_meta import (
     read_tensor_meta,
-    write_tensor_meta,
     update_tensor_meta_with_array,
     validate_tensor_meta,
+    write_tensor_meta,
 )
-from hub.core.meta.index_map import read_index_map, write_index_map
-from hub.util.keys import get_tensor_meta_key, get_index_map_key
-from hub.util.array import normalize_and_batchify_array_shape
 from hub.core.typing import StorageProvider
+from hub.util.array import normalize_and_batchify_array_shape
+from hub.util.dataset import get_compressor
 from hub.util.exceptions import (
     DynamicTensorNumpyError,
+    ImageReadError,
     TensorAlreadyExistsError,
+    TensorDoesNotExistError,
     TensorInvalidSampleShapeError,
     TensorMetaMismatchError,
-    TensorDoesNotExistError,
+    WrongMetadataError,
```

Comment:
this exception should not exist, please use `TensorMetaMismatchError`

Reply:
actually, it should exist separately I apologize. however, could you rename it to `SampleCorruptedError`